portingdb/htmlreport.py

```diff
@@ -479,34 +480,29 @@ def howto():
     db = current_app.config['DB']()
     query = queries.packages(db)
 
-    # Count the idle and blocked packages
-    idle_query, query = queries.split(query, tables.Package.status == 'idle')
-    idle_len = idle_query.count()
-    blocked_query, query = queries.split(query, tables.Package.status == 'blocked')
+    # Count the blocked packages
+    blocked_query = query.filter(tables.Package.status == 'blocked')
     blocked_len = blocked_query.count()
 
+    # Get all the idle packages
+    idle_query = query.filter(tables.Package.status == 'idle')
+    idle = list(idle_query)
+    idle_len = len(idle)
+
     # Get all the mispackaged packages
-    mispackaged_query, query = queries.split(query, tables.Package.status == 'mispackaged')
+    mispackaged_query = query.filter(tables.Package.status == 'mispackaged')
     mispackaged = list(mispackaged_query)
 
+    # Pick an idle package at random
+    random_idle = random.choice(idle)
+
     # Pick a mispackaged package at random
     random_mispackaged = random.choice(mispackaged)
 
-    # Fedora status: Mispackaged
-    query = db.query(tables.Status)
-    query = query.filter(tables.Status.ident == 'mispackaged')
-    query = query.join(tables.CollectionPackage)
-    query = query.filter(
-            tables.CollectionPackage.collection_ident == 'fedora')
-    mispackaged_status = query.first()
-
-    # Upstream status: Released
+    # Status objects
     query = db.query(tables.Status)
-    query = query.filter(tables.Status.ident == 'released')
-    query = query.join(tables.CollectionPackage)
-    query = query.filter(
-            tables.CollectionPackage.collection_ident == 'upstream')
-    released_status = query.first()
+    mispackaged_status = query.get('mispackaged')
+    released_status = query.get('released')
```

Comment:
i.e. the `tables.Status` is the same, there are `tables.CollectionStatus` but those just add a collection-dependent description.